includes/class-wc-ajax.php

```diff
@@ -1306,6 +1306,13 @@ public static function product_ordering() {
 			$index ++;
 			$menu_orders[ $id ] = $index;
 			$wpdb->update( $wpdb->posts, array( 'menu_order' => $index ), array( 'ID' => $id ) );
+			
+			/**
+			 * When a single product has gotten it's ordering updated.
+			 * $id The product ID
+			 * $index The new menu order
+			*/
+			do_action( 'woocommerce_after_single_product_ordering', $id, $index );
```

Comment:
Well its already possible to easily target all the other use cases, so as updated ordering via a POST. Besides, a do_action without any listeners is negligible in performance loss, as you know.

We could probably catch this at much later hook (maybe a database one), but performance should always be a concern, and listening to such a hook would generate a lot of irrelevant noise.